lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php

```diff
@@ -121,39 +122,64 @@ public function convertException($message, DriverException $exception)
         return new Exception\DriverException($message, $exception);
     }
 
+
     /**
      * {@inheritdoc}
      */
     public function createDatabasePlatformForVersion($version)
     {
-        if ( ! preg_match('/^(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?/', $version, $versionParts)) {
-            throw DBALException::invalidPlatformVersionSpecified(
-                $version,
-                '<major_version>.<minor_version>.<patch_version>'
-            );
-        }
+        // Detect vendor and version
 
         if (false !== stripos($version, 'mariadb')) {
-            return $this->getDatabasePlatform();
-        }
 
-        $majorVersion = $versionParts['major'];
-        $minorVersion = isset($versionParts['minor']) ? $versionParts['minor'] : 0;
-        $patchVersion = isset($versionParts['patch']) ? $versionParts['patch'] : null;
+            // Hack to ignore things like '5.5.5-10.2.8-MariaDB-10.2.8+maria~xenial-log'
+            $version = str_replace('5.5.5-', '', $version);
 
-        if ('5' === $majorVersion && '7' === $minorVersion && null === $patchVersion) {
-            $patchVersion = '9';
-        }
+            if (!preg_match('/^(mariadb-)?(?P<major>\d+)\.(?P<minor>\d+)\.(?P<patch>\d+)/', strtolower($version), $versionParts)) {
```

Comment:
This entire block needs to be moved to a `private` method

Reply:
Nice, just added `AbstractMySQLDriver::getMariaDbMysqlVersionNumber($version)`.